Mathlib/Algebra/MonoidAlgebra/Equiv.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2022 Yuyang Zhao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yuyang Zhao
+-/
+
+import Mathlib.Algebra.Algebra.Tower
+import Mathlib.Algebra.MonoidAlgebra.Basic
+
+/-!
+TODO
+-/
+
+noncomputable section
+
+namespace AddMonoidAlgebra
+
+@[simps]
+def ringHomCongrLeft {R S G : Type _} [Semiring R] [Semiring S] [AddZeroClass G] (f : R →+* S) :
+    AddMonoidAlgebra R G →+* AddMonoidAlgebra S G :=
+  { Finsupp.mapRange.addMonoidHom f.toAddMonoidHom with
```

Comment:
If you use one of the `lift` functions can you avoi needing to prove the `map_mul'`?